tidy3d/web/api/container.py

```diff
@@ -719,14 +731,18 @@ def download(self, path_dir: str = DEFAULT_DATA_DIR) -> None:
 
         self.to_file(self._batch_path(path_dir=path_dir))
 
-        for task_name, job in self.jobs.items():
-            job_path = self._job_data_path(task_id=job._task_id, path_dir=path_dir)
+        # TODO: progressbar
+        # TODO: multi-threading
+        with ThreadPoolExecutor(max_workers=MAX_WORKERS) as executor:
+            for task_name, job in self.jobs.items():
+                job_path = self._job_data_path(task_id=job.task_id, path_dir=path_dir)
 
-            if "error" in job.status:
-                log.warning(f"Not downloading '{task_name}' as the task errored.")
-                continue
+                if "error" in job.status:
+                    log.warning(f"Not downloading '{task_name}' as the task errored.")
+                    continue
 
-            job.download(path=job_path)
+                print(f"downloading '{task_name}'")
+                future = executor.submit(lambda: job.download(path=job_path))
```

Comment:
how to wait for all files get download successfully?

Reply:
Good point, so I want to basically block the return of this function until all futures have downloaded? I think through `ThreadPoolExecutor` I can check the status of the queue and maybe return if it is empty. I'll look into this.

Reply:
It looks like the `ThreadPoolExecutor` context manager already waits to continue until all `futures` have been completed. So it should be fine.